tools/xnvme.c

```diff
@@ -623,6 +623,45 @@ sub_library_info(struct xnvmec *XNVME_UNUSED(cli))
 	return err;
 }
 
+static int
+sub_dsm(struct xnvmec *cli)
+{
+	// We can only define one range in CLI
+	uint32_t nr = 1;
+	struct xnvme_spec_dsm_range dsm_range;
```

Comment:
As a data-payload, then the ``dsm_range`` should be allocated by ``xnvme_buf_alloc()`` not on the stack.

Reply:
Fixed!